annotator-core/src/main/java/edu/ucr/cs/riple/core/metadata/index/Error.java

```diff
@@ -166,4 +184,39 @@ public int hashCode() {
   public String toString() {
     return "Type='" + messageType + '\'' + ", message='" + message + '\'';
   }
+
+  /**
+   * Collects all resolving fixes in given collection of errors as an immutable set.
+   *
+   * @param errors Collection of errors.
+   * @return Immutable set of fixes which can resolve all given errors.
+   */
+  public static ImmutableSet<Fix> getResolvingFixesOfErrors(Collection<Error> errors) {
+    // Each error has a set of resolving fixes and each fix has a set of reasons as why the fix has
+    // been suggested. The final returned set of fixes should contain all the reasons it has been
+    // suggested across the given collection. Map below stores all the set of reasons each fix is
+    // suggested in the given collection.
+
+    // Collect all reasons each fix is suggested across the given collection.
+    Map<Fix, Set<String>> fixReasonsMap = new HashMap<>();
+    errors.stream()
+        .flatMap(error -> error.resolvingFixes.stream())
+        .forEach(
+            fix -> {
+              if (fixReasonsMap.containsKey(fix)) {
+                fixReasonsMap.get(fix).addAll(fix.reasons);
+              } else {
+                fixReasonsMap.put(fix, new HashSet<>(fix.reasons));
+              }
+            });
+
+    ImmutableSet.Builder<Fix> builder = ImmutableSet.builder();
+    for (Fix key : fixReasonsMap.keySet()) {
+      // To avoid mutating fixes stored in the given collection, we create new instances.
```

Comment:
```suggestion
      // To avoid mutating fixes stored in the given collection, we create new instances.
      // which contain the full set of reasons.
```